enterprise_access/apps/content_assignments/models.py

```diff
@@ -784,9 +784,23 @@ def annotate_dynamic_fields_onto_queryset(cls, queryset):
                     error_reason__isnull=True,
                     completed_at__isnull=False,
                 )
+            ),
+            # ... or if they have an errored notification.
+            has_errored_notification=Exists(
+                LearnerContentAssignmentAction.objects.filter(
+                    assignment=OuterRef('uuid'),
+                    action_type=AssignmentActions.NOTIFIED,
+                    error_reason__isnull=False,
+                )
             )
         ).annotate(
             learner_state=Case(
+                When(
+                    Q(state=LearnerContentAssignmentStateChoices.ALLOCATED) &
+                    Q(has_errored_notification=True) &
+                    Q(has_notification=False),
+                    then=Value(AssignmentLearnerStates.FAILED),
```

Comment:
I think this should work (not creating a new learner_state specifically for this case), but I'm not 100% sure. The ticket description recommends a new learner state, but just wanted to make sure we confirmed the frontend is okay with this version.

Reply:
Yup, the frontend already respects the `FAILED` value, that's how we get the "Failed: Bad Email" in the admin-portal screenshot in the description.